src/app/pages/system/general-settings/support/support.component.ts

```diff
@@ -150,10 +150,19 @@ export class SupportComponent implements OnInit {
   }
 
   updateProductionStatus(event: MatCheckboxChange): void {
-    let request$: Observable<boolean | SetProductionStatusDialogResult> = of(true);
+    let request$: Observable<boolean | SetProductionStatusDialogResult> = of(false);
     if (event.checked) {
       request$ = request$.pipe(
-        switchMap(() => this.matDialog.open(SetProductionStatusDialogComponent).afterClosed()),
+        switchMap(() => this.matDialog.open(SetProductionStatusDialogComponent).afterClosed().pipe(
+          map((confirmed) => {
+            if (confirmed) {
```

Comment:
The `map` operator seems unnecessary here since we don't change the resulting output of the stream, so the `tap` operator may fit better to produce side effects. If there was an intention to use especially `map,` it is okay too.
```
tap((confirmed) => { 
  if (!confirmed) { 
    this.isProduction = false; 
    this.cdr.markForCheck();
  }
})
```